components/scream/src/share/io/scorpio_input.hpp

```diff
@@ -96,28 +97,52 @@ class AtmosphereInput
     m_field_repo = repo;
     m_gm         = gm;
   }
+  AtmosphereInput(const ekat::Comm& comm, const std:: string grid_name,
+                  const std::shared_ptr<const GridsManager>& gm)
+  {
+    m_comm       = comm;
+    m_grid_name  = grid_name;
+    m_gm         = gm;
+  }
 
   // Main Functions
   void pull_input();
-  void pull_input(const std::string name, view_type view_out);
+  void pull_input(const std::string& name, view_type view_out);
+  void pull_input(const std::string& filename, const std::string& var_name, const std::vector<std::string>& var_dims,
+                  const bool has_columns, const std::vector<int>& dim_lens, const int padding, Real* data);
+
+  // Determine padding from the type of the variable.
```

Comment:
It's worth pointing out in a comment that this fcn _expects_ `var_dims` to be the "physical" dimensions of the field, as opposed to the "array" dimensions. That is, `var_dims` are the dimensions it would have if ValueType=Real (no packing).

Reply:
done, I've added a comment.